dashboard/src/components/AppView/AppControls/RollbackButton/RollbackButton.test.tsx

```diff
@@ -0,0 +1,146 @@
+import { shallow, ShallowWrapper } from "enzyme";
+import context from "jest-plugin-context";
+import * as React from "react";
+import * as ReactModal from "react-modal";
+import { IAppRepository, IChartVersion, IRelease } from "shared/types";
+import RollbackButton from ".";
+import SelectRepoForm from "../../../../components/SelectRepoForm";
+import RollbackDialog from "./RollbackDialog";
+
+const defaultProps = {
+  app: {} as IRelease,
+  rollbackApp: jest.fn(),
+  getChartVersion: jest.fn(),
+  loading: false,
+  repos: [],
+  repo: {} as IAppRepository,
+  kubeappsNamespace: "kubeapps",
+  fetchRepositories: jest.fn(),
+  checkChart: jest.fn(),
+};
+
+it("stores the chart name and version in the state", () => {
+  const wrapper = shallow(<RollbackButton {...defaultProps} />);
+  wrapper.setProps({
```

Comment:
Calling `setProps` forces the button to update right after shallowing them. If you are not testing this update behaviour, it's better to avoid updating the component right after shallowing it. You can use the spread operator to modify the `defaultProps`. For example:

```
const appProps = {
  ...defaultProps,
  app: {
    // ...
  }
}

const wrapper = shallow(<RollbackButton {...appProps} />);
```